common/src/main/java/org/valkyrienskies/mod/mixin/feature/ai/node_evaluator/SwimNodeEvaluatorMixin.java

```diff
@@ -0,0 +1,188 @@
+                        final BlockPos groundPos = new BlockPos(x, y, z);
+                        fluidState[0] = finalLevel.getFluidState(groundPos);
+                    });
+            }
+        }
+        return fluidState[0];
+    }
+
+    @Redirect(
+        at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/level/BlockGetter;getBlockState(Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/block/state/BlockState;"),
+        method = "getBlockPathType(Lnet/minecraft/world/level/BlockGetter;III)Lnet/minecraft/world/level/pathfinder/BlockPathTypes;"
+    )
+    private BlockState getBlockStateRedirectPathType(final BlockGetter instance, final BlockPos blockPos) {
+        final BlockState[] blockState = {instance.getBlockState(blockPos)};
+        if (instance instanceof PathNavigationRegion && blockState[0].isAir() && VSGameConfig.SERVER.getAiOnShips()) {
+            final Level level = ((PathNavigationRegionAccessor) instance).getLevel();
+
+            final double origX = blockPos.getX();
+            final double origY = blockPos.getY();
+            final double origZ = blockPos.getZ();
+            VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                origY, origZ, 1,
+                (x, y, z) -> {
+                    final BlockPos groundPos = new BlockPos(x, y, z);
+                    blockState[0] = level.getBlockState(groundPos);
+                });
+        }
+        return blockState[0];
+    }
+
+    @Redirect(
+        at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/level/block/state/BlockState;isPathfindable(Lnet/minecraft/world/level/BlockGetter;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/pathfinder/PathComputationType;)Z"),
+        method = "getBlockPathType(Lnet/minecraft/world/level/BlockGetter;III)Lnet/minecraft/world/level/pathfinder/BlockPathTypes;"
+    )
+    private boolean isPathFindableRedirectPathType(final BlockState instance, final BlockGetter blockGetter,
+        final BlockPos blockPos,
+        final PathComputationType pathComputationType) {
+        final boolean[] isPathFindable = {instance.isPathfindable(blockGetter, blockPos, pathComputationType)};
+        if (!isPathFindable[0] && VSGameConfig.SERVER.getAiOnShips()) {
+            Level level = null;
+            if (blockGetter instanceof PathNavigationRegion) {
+                level = ((PathNavigationRegionAccessor) blockGetter).getLevel();
+            } else if (blockGetter instanceof Level) {
+                level = (Level) blockGetter;
+            }
+            if (level != null) {
+                final double origX = blockPos.getX();
+                final double origY = blockPos.getY();
+                final double origZ = blockPos.getZ();
+                final Level finalLevel = level;
+                VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                    origY, origZ, 1,
+                    (x, y, z) -> {
```

Comment:
Why are you using `transformToNearbyShipsAndWorld` if it only uses the last x,y,z that it gets and ignores the rest?